agent/config/parse.go

```diff
@@ -206,6 +206,39 @@ func parseTaskMetadataThrottles() (int, int) {
 	return steadyStateRate, burstRate
 }
 
+func parseContainerInstanceTags(errs []error) (map[string]string, []error) {
+	var containerInstanceTags map[string]string
+	containerInstanceTagsConfigString := os.Getenv("ECS_CONTAINER_INSTANCE_TAGS")
+
+	// If duplicate keys exist, the value of the key will be the value of latter key.
+	err := json.Unmarshal([]byte(containerInstanceTagsConfigString), &containerInstanceTags)
+	if containerInstanceTagsConfigString != "" {
+		if err != nil {
+			wrappedErr := fmt.Errorf("Invalid format for ECS_CONTAINER_INSTANCE_TAGS. Expected a json hash: %v", err)
+			seelog.Error(wrappedErr)
+			errs = append(errs, wrappedErr)
+		}
+	}
+
+	for tagKey, tagValue := range containerInstanceTags {
+		seelog.Debugf("Setting instance tag %v: %v", tagKey, tagValue)
+	}
+
+	return containerInstanceTags, errs
+}
+
+func parseContainerInstancePropagateTags() ContainerInstancePropagateTagsType {
+	containerInstancePropagateTagsString := os.Getenv("ECS_CONTAINER_INSTANCE_PROPAGATE_TAGS")
```

Comment:
I feel like `ECS_CONTAINER_INSTANCE_PROPAGATE_TAGS` should be a list of strings rather than one, if we want to support multiple tag types in the future and not break backwards compatibility.

Reply:
I think customer only has one option - either ask Agent to propagate the tags or ask backend to propagate the tags, customer will get the same tags either way, so supporting both of them seems to do duplicate work.

Reply:
Yes, it can be other values, but even if there are other possible values, customer can only use one of them. For example, if we have `backend` as one of the values in the future, customer only needs `ECS_CONTAINER_INSTANCE_PROPAGATE_TAGS` to be either `ec2_instance` or `backend`, because we will get the same set of tags when we ask backend to propagate or we let Agent propagate tags.